example/src/main/java/com/stripe/example/activity/AndroidPayActivity.java

```diff
@@ -0,0 +1,311 @@
+        mFragmentContainer.setVisibility(View.GONE);
+        mChangeDetailsContainer.setVisibility(View.GONE);
+    }
+
+    @Override
+    protected void addBuyButtonWalletFragment(@NonNull SupportWalletFragment walletFragment) {
+        FragmentTransaction buttonTransaction = getSupportFragmentManager().beginTransaction();
+        buttonTransaction.add(R.id.container_android_pay_button, walletFragment).commit();
+    }
+
+    @Override
+    protected void addConfirmationWalletFragment(@NonNull SupportWalletFragment walletFragment) {
+        mChangeDetailsContainer.setVisibility(View.VISIBLE);
+        FragmentTransaction confirmationTransaction =
+                getSupportFragmentManager().beginTransaction();
+        confirmationTransaction.setTransition(FragmentTransaction.TRANSIT_FRAGMENT_OPEN);
+        confirmationTransaction.replace(R.id.container_fragment_confirmation, walletFragment).commit();
+    }
+
+    @NonNull
+    @Override
+    protected WalletFragmentStyle getWalletFragmentConfirmationStyle() {
+        return new WalletFragmentStyle()
+                .setMaskedWalletDetailsLogoImageType(WalletFragmentStyle.LogoImageType.ANDROID_PAY)
+                .setStyleResourceId(R.style.SampleTheme)
+                .setMaskedWalletDetailsTextAppearance(android.R.style.TextAppearance_DeviceDefault_Medium)
+                .setMaskedWalletDetailsHeaderTextAppearance(android.R.style.TextAppearance_DeviceDefault_Large);
+    }
+
+    @Override
+    protected void onChangedMaskedWalletRetrieved(@Nullable MaskedWallet maskedWallet) {
+        super.onChangedMaskedWalletRetrieved(maskedWallet);
+        if (maskedWallet == null) {
+            return;
+        }
+        mPossibleConfirmedMaskedWallet = maskedWallet;
+        mChangeDetailsContainer.setVisibility(View.GONE);
+        mConfirmDetailsContainer.setVisibility(View.VISIBLE);
+        updatePaymentInformation(mPossibleConfirmedMaskedWallet);
+    }
+
+    @Override
+    protected void onMaskedWalletRetrieved(@Nullable MaskedWallet maskedWallet) {
+        if (maskedWallet != null) {
+            mPossibleConfirmedMaskedWallet = maskedWallet;
+
+            updatePaymentInformation(mPossibleConfirmedMaskedWallet);
+
+            mChangeDetailsContainer.setVisibility(View.GONE);
+            mConfirmDetailsContainer.setVisibility(View.VISIBLE);
+        }
+    }
+
+    @Override
+    protected void onStripePaymentSourceReturned(
+            FullWallet wallet,
+            StripePaymentSource paymentSource) {
+        super.onStripePaymentSourceReturned(wallet, paymentSource);
+        mProgressDialogController.finishProgress();
+
+        String cardDetails = null;
+        if (wallet.getInstrumentInfos() != null && wallet.getInstrumentInfos().length > 0) {
+            InstrumentInfo info = wallet.getInstrumentInfos()[0];
+            cardDetails = info.getInstrumentDetails();
+        }
+
+        String id = paymentSource.getId();
```

Comment:
Probably out of the scope of this PR, but I think in addition to displaying the created tokens in a list, we should send the token to an example backend to create a charge, as you do in the emoji example app. 

For now, can we add a comment to that effect? "Send the ID of the payment source to your backend to complete the charge"